Examples/Detectors/ContextualDetector/src/PrealignedDetector.cpp

```diff
@@ -0,0 +1,169 @@
+    -> std::pair<TrackingGeometryPtr, ContextDecorators> {
+  // --------------------------------------------------------------------------------
+  DetectorElement::ContextType nominalContext;
+
+  auto buildLevel = vm["geo-generic-buildlevel"].template as<size_t>();
+  // set geometry building logging level
+  Acts::Logging::Level surfaceLogLevel =
+      Acts::Logging::Level(vm["geo-surface-loglevel"].template as<size_t>());
+  Acts::Logging::Level layerLogLevel =
+      Acts::Logging::Level(vm["geo-layer-loglevel"].template as<size_t>());
+  Acts::Logging::Level volumeLogLevel =
+      Acts::Logging::Level(vm["geo-volume-loglevel"].template as<size_t>());
+
+  bool buildProto =
+      (vm["mat-input-type"].template as<std::string>() == "proto");
+
+  /// return the generic detector - with aligned context decorator
+  TrackingGeometryPtr aTrackingGeometry =
+      ActsExamples::Generic::buildDetector<DetectorElement>(
+          nominalContext, detectorStore, buildLevel, std::move(mdecorator),
+          buildProto, surfaceLogLevel, layerLogLevel, volumeLogLevel);
+
+  Acts::Logging::Level decoratorLogLevel =
+      Acts::Logging::Level(vm["prealign-loglevel"].template as<size_t>());
+
+  // Let's create a reandom number service
+  ActsExamples::RandomNumbers::Config randomNumberConfig;
+  randomNumberConfig.seed = vm["prealign-seed"].template as<size_t>();
+  auto randomNumberSvc =
+      std::make_shared<ActsExamples::RandomNumbers>(randomNumberConfig);
+
+  // Alignment decorator service
+  Decorator::Config pagcsConfig;
+  pagcsConfig.detectorStore = detectorStore;
+  pagcsConfig.nIovs = vm["prealign-iovs"].template as<size_t>();
+  pagcsConfig.randomNumberSvc = randomNumberSvc;
+
+  // Now create the alignment decorator
+  ContextDecorators aContextDecorators = {std::make_shared<Decorator>(
+      pagcsConfig,
+      Acts::getDefaultLogger("PrealignedDecorator", decoratorLogLevel))};
+
+  bool firstIovNominal = vm["prealign-firstnominal"].template as<bool>();
+
+  // Create an local generator for the preContext
+  size_t max = std::numeric_limits<size_t>::max();
+  ActsExamples::WhiteBoard preStore(Acts::getDefaultLogger(
+      "PreEventStore#" + std::to_string(max), decoratorLogLevel));
+  ActsExamples::AlgorithmContext preContext(max, max, preStore);
+  ActsExamples::RandomEngine rng = randomNumberSvc->spawnGenerator(preContext);
+  std::normal_distribution<double> gauss(0., 1.);
+
+  // The misalingments
+  double sigmaIp = vm["prealign-sigma-iplane"].template as<double>();
+  double sigmaOp = vm["prealign-sigma-oplane"].template as<double>();
+  double sigmaIr = vm["prealign-sigma-irot"].template as<double>();
+  double sigmaOr = vm["prealign-sigma-orot"].template as<double>();
+  double gSigmaX = sigmaIp * Acts::UnitConstants::um;
+  double gSigmaY = sigmaIp * Acts::UnitConstants::um;
+  double gSigmaZ = sigmaOp * Acts::UnitConstants::um;
+  double aSigmaX = sigmaOr * 0.001;  // millirad
```

Comment:
Aren't the `sigmaOr` and `sigmaIr` supposed to in units of `mrad` already? The same case in the `AlignedDetector.cpp`.

Reply:
Oh, could be - will check.